API/Controllers/UsersController.cs

```diff
@@ -1,37 +1,103 @@
 ﻿using API.Models;
+using API.Models.AccountModels;
 using API.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using System.Threading.Tasks;
 
 namespace API.Controllers
 {
     [Route("api/v1/[controller]/[action]")]
     [Produces(ApiContentType)]
     [ApiController]
+    [Authorize]
     public class UsersController : ApiControllerBase
     {
-        public UsersController(ApiContext context, Iserializer serializer, IErrorFormatter errorFormatter) :
-            base(context, serializer, errorFormatter) { }
+        private readonly UserManager<UserDTO> _userManager;
+        private readonly SignInManager<UserDTO> _signInManager;
+        private readonly JwtService _jwt;
+        public UsersController(
+            ApiContext context,
+            Iserializer serializer,
+            IErrorFormatter errorFormatter,
+            IConfiguration config,
+            UserManager<UserDTO> userManager,
+            SignInManager<UserDTO> signInManager) :
+            base(context, serializer, errorFormatter)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _jwt = new JwtService(config);
+        }
 
         [HttpPost]
-        public IActionResult Signup(UserDTO user)
+        [AllowAnonymous]
+        public async Task<IActionResult> Signup(RegisterModel model)
         {
             if (!IsValidApiRequest())
             {
                 return ApiBadRequest("Invalid headers!");
             }
 
-            return NotImplementedError("Signup Not Implemented!");
+            if (ModelState.IsValid)
+            {
+                var user = new UserDTO { UserName = model.Username ?? model.Email, Email = model.Email };
+                var result = await _userManager.CreateAsync(user, model.Password);
+                if (result.Succeeded)
+                {
+                    return Created("", new
+                    {
+                        token = _jwt.GenerateSecurityToken(new User()
+                        {
+                            Username = user.UserName,
+                            Email = user.Email
+                        })
+                    });
+                }
+
+                return BadRequest(result);
```

Comment:
Also, I receive these warnings in console:
```
warn: Microsoft.AspNetCore.Identity.UserManager[14]
      User e1612359-a811-4c21-81e2-360136b228d1 password validation failed: PasswordTooShort;PasswordRequiresNonAlphanumeric;PasswordRequiresDigit;PasswordRequiresUpper.
warn: Microsoft.AspNetCore.Identity.UserManager[14]
      User af0e5455-c755-4640-b9f4-70185cd04a08 password validation failed: PasswordRequiresNonAlphanumeric.
warn: Microsoft.AspNetCore.Identity.UserManager[14]
      User edc5c756-9ab6-47e0-a4a3-33d24bdd7da1 password validation failed: PasswordTooShort;PasswordRequiresDigit;PasswordRequiresUpper.
```
This information is completely irrelevant to be logged in console.

Reply:
I've suppressed it for signup, but found no good way to do so for login method.